src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -98,38 +106,56 @@ public Object instantiate() throws ConstructionException {
         } catch (IllegalAccessException e) {
             throw new ConstructionException(ctor, "Access violation on " + ctor, e);
         }
+        finally {
+            mdcContext.finish();
+        }
 
         // satisfy dependencies in the order of the list, which was
         // prepared to comply with JSR 330
         Map<Method, InjectionArgs> settersAndArguments = new HashMap<Method, InjectionArgs>();
         for (Desire d: desires) {
+            Object value;
+            FieldInjectionPoint fd;
+            Field field;
+            LogContext mdcContextInjectionPointFieldValue;
             if (d.getInjectionPoint() instanceof FieldInjectionPoint) {
-                FieldInjectionPoint fd = (FieldInjectionPoint) d.getInjectionPoint();
-                Object value = checkNull(fd, providers.get(d).instantiate());
-                Field field = fd.getMember();
-
+                mdcContextInjectionPointFieldValue = LogContext.create();
+                fd = (FieldInjectionPoint) d.getInjectionPoint();
+                value = checkNull(fd, providers.get(d).instantiate());
```

Comment:
This is the key line that must be inside the `try`, after the `put` calls on the context.

We might need another level of `try` statements.

We might want to consider factoring this method into several methods to reduce its complexity.